src/main/java/com/direwolf20/buildinggadgets/common/util/helpers/SortingHelper.java

```diff
@@ -55,5 +65,46 @@
 
     }
 
+    public static class RenderSorter {
```

Comment:
Whats all this for?

Reply:
I want something that can hold the blocks both in the way they come and in the sorted order, without having to worry about managing the list myself. 
I'm still not sure why you said you need it sorted in the reverse Direction (I intend on trying without), but we should be able to manage that in reverse Direction.

I'll also write a sub-class using uniform Sort, to test the improvements by using an expected O(n) algorithm.

Reply:
The reverse shit was all about me misunderstanding how OpenGl layers renders. Now that we've moved the render (1.12) to a new for loop the issue is no longer presented and the minor changes to the IBuildingMode has resolved all the sorting issues Dire was having when using the mod. 

With this info. Is this still needed?

Reply:
Yes, because your changes to the BuildingMode, don't affect the IBuildView provided by the copy-paste Gadget. 
Also I intended on removing the remaining code of BlockSorter (and the slower implementation), once I find out which sorting algorithm is faster...